### DISCUSSION TITLE
help with static files in bento endpoint

### DISCUSSION THREAD

**Discussion by psybo7** [bentoml/BentoML#1325] [Q&A]
Hi!
I am trying to serve a bentoML prediction service as a Kubernetes deployment. I am using bentoML 0.10.
Besides the deployment, I defined a service and an ingress (my ingress controller is NginX).
It all seems fine but there is no way I can reach the swagger interface at the specified URL. 
If I try to do it with the browser I only get a blank page (which, if inspected, shows the swagger HTML structure).
If I try to curl the link I only get 404 error messages.

```
HTTP/1.1 404 NOT FOUND
Date: Mon, 14 Dec 2020 15:00:10 GMT
Content-Type: text/html; charset=utf-8
Content-Length: 232
Connection: keep-alive

<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<title>404 Not Found</title>
<h1>Not Found</h1>
<p>The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again.</p>
```

Apparently the system cannot reach swagger_static/swagger-ui.css and swagger_static/swagger-ui-bundle.js
On the contrary, everything works fine if I just try to locally port forward the deployment (eg. kubectl port-forward bentopod 5000:5000).
Do you have any suggestions to help me solve this problem? Should I add some extra code to my bento Api?
Thank you a lot,
Matt

**Comment by parano** [comment]
Thanks for reporting this issue @matt985,  cc @yubozhao could you take a look? It seems a case we should consider when addressing https://github.com/bentoml/BentoML/issues/1183 in 
https://github.com/bentoml/BentoML/pull/1207

**Reply by yubozhao** [reply]
Yeah, I think this case should be part of that PR, and I will look into this problem.

I created #1328 to track this.

**Comment by psybo7** [comment]
Hi!
I have an update on this issue. 
We eventually managed to serve the prediction service as a kubernetes deployment with a NginX ingress controller. One of the problems was the rewrite rule configuration of the ingress.
Specifically, we changed the field in "spec.rules.host.http.paths.path" from the standard "/classifier-name(/|$)(.*)" to "/classifier-name".
Even if a bit strange (in combination with: nginx.ingress.kubernetes.io/rewrite-target: /$2), it works.
If I curl the service I can get the prediction:

```
curl -i \
--header "Content-Type: application/json" \
--request POST \
--data '[[5.1, 3.5, 1.4, 0.2]]' \
http://host/classifier-name/predict
[0]
```


I can also get to the Swagger UI. Unfortunately this interface does not work. Apparently it calls the wrong endpoint:

`curl -X POST "http://host/predict" -H "accept: */*" -H "Content-Type: application/json" -d "[[5.1,3.5,1.4,0.2]]"`

Instead of:

`curl -X POST "http://host/classifier-name/predict" -H "accept: */*" -H "Content-Type: application/json" -d "[[5.1,3.5,1.4,0.2]]"`

**Reply by jcriddle-nci** [reply]
Has any solution come out of this? We are seeing this same problem and are struggling to find a good solution

**Reply by yubozhao** [reply]
@jcriddle-nci I want to make sure, are you trying to access the swagger UI and make example prediction from there and failed? Could you share me with more about the use cases that involves with swagger UI beside developing?

There is a PR #1411 that enable customized swagger endpoints.
